src/java/fr/paris/lutece/portal/web/download/DownloadServlet.java

```diff
@@ -0,0 +1,56 @@
+package fr.paris.lutece.portal.web.download;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import fr.paris.lutece.portal.business.file.File;
+import fr.paris.lutece.portal.service.download.AbstractFileDownloadProvider;
+import fr.paris.lutece.portal.service.download.IFileDownloadProvider;
+import fr.paris.lutece.portal.service.message.SiteMessageException;
+import fr.paris.lutece.portal.service.message.SiteMessageService;
+import fr.paris.lutece.portal.service.util.AppPathService;
+
+public class DownloadServlet extends HttpServlet
+{
+    private static final long serialVersionUID = 6622358100579620819L;
+    private static final String MESSAGE_UNKNOWN_PROVIDER = "portal.file.download.provider.unknown";
+    private static final String MESSAGE_UNKNOWN_FILE = "portal.file.download.file.unknown";
+    
+    @Override
+    protected void doGet( HttpServletRequest request, HttpServletResponse response ) throws ServletException, IOException
+    {
+        File file = null;
+        try
+        {
+            IFileDownloadProvider provider = AbstractFileDownloadProvider.findProvider( request.getParameter( AbstractFileDownloadProvider.PARAM_PROVIDER ) );
+            if ( provider == null )
+            {
+                SiteMessageService.setMessage( request, MESSAGE_UNKNOWN_PROVIDER );
+            }
+            file = provider.getFile( request );
+            if ( file == null || file.getPhysicalFile( ) == null )
+            {
+                SiteMessageService.setMessage( request, MESSAGE_UNKNOWN_FILE );
+            }
+        }
+        catch( SiteMessageException e )
+        {
+            response.sendRedirect( AppPathService.getSiteMessageUrl( request ) );
+        }
+        
+        if ( file != null )
+        {
+            try ( OutputStream outputStream = response.getOutputStream( ) )
+            {
+                response.setContentType( file.getMimeType( ) );
+                response.setHeader( "Content-Disposition", "attachment; filename=\"" + file.getTitle( ) + "\";" );
+                outputStream.write( file.getPhysicalFile( ).getValue( ) );
```

Comment:
What if the file is large ? Doesn't this make easy to DOS the server if I have a valid link to a large file by accessing it multiple times until an OOM error occurs ?